### PR TITLE
fix(selector): removing an item sometimes clear the current selection

### LINKED ISSUE #17010: TabView's tab selection does not work properly in Desktop
### Current behavior

On the Desktop head (Skia WPF), when having two opened tabs (one selected and one not), closing the unselected tab causes the previously selected tab to be deselected.

### Expected behavior

When closing the unselected tab, the previously selected tab has to stay selected.

### How to reproduce it (as minimally and precisely as possible)

 1. Open the attached solution and compile it.
 2. Execute it in Desktop head.
 3. With the second tab selected, close the first one.
 4. Then the second tab will be automatically deselected.
[TabViewSelectionBugWPF.zip](https://github.com/user-attachments/files/15593178/TabViewSelectionBugWPF.zip)

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

Uno.WinUI 5.2.139

### Affected platforms

Skia (WPF)

### IDE

Visual Studio 2022

### IDE version

17.10.1

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #17010

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/Primitives/Selector.cs [hunk 1]
```diff
@@ -683,14 +683,6 @@ private void OnItemsChanged(NotifyCollectionChangedEventArgs e)
...
 					}
 				}
-				else if (iVCE.CollectionChange == CollectionChange.ItemRemoved)
-				{
-					// If the removed item is the currently selected one, Set SelectedIndex to -1
-					if ((int)iVCE.Index == SelectedIndex)
-					{
-						ResetIndexIfNeeded();
-					}
-				}
```
Comment (ramezgerges) [inline src/Uno.UI/UI/Xaml/Controls/Primitives/Selector.cs:693]: `OnItemsSourceSingleCollectionChanged` (which is the only call site for this function) already calls `ResetIndexIfNeeded`, so calling it again here causes the bug.

Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/Primitives/Selector.cs:693]: Let's make sure that the tests from https://github.com/unoplatform/uno/pull/9340, https://github.com/unoplatform/uno/pull/6742 and https://github.com/unoplatform/uno/pull/4096 are still passing.

Comment (ramezgerges) [reply src/Uno.UI/UI/Xaml/Controls/Primitives/Selector.cs:693]: CI is green and the manual tests from #4096 look fine.

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.3

**Pull request merged**